onnxruntime/core/providers/openvino/backends/basic_backend.cc

```diff
@@ -313,13 +313,28 @@ void BasicBackend::PopulateConfigValue(ov::AnyMap& device_config) {
   }
 }
 
-void BasicBackend::EnableCaching() {
+void BasicBackend::EnableCaching(ov::AnyMap& device_config) {
   // cache_dir argument has no effect when working with an embed-mode EPContext Graph
   if (subgraph_context_.is_ep_ctx_graph) return;
 
   if (!session_context_.cache_dir.empty() && !session_context_.so_context_enable) {
     LOGS_DEFAULT(INFO) << log_tag << "Enables Caching";
+    
     OVCore::Get()->SetCache(session_context_.cache_dir.string());
+    config_t& load_config = session_context_.load_config;
```

Comment:
I see that you are setting the values explicitly, can we not directly pass below & let ov toolkit take care of it. In my opinion this change is not needed, as we are delegating all the heavy lifting & setting of ov::AnyMap device_config  properties to OV toolkit internally going ahead

```
{
 "GPU": {
    "CACHE_MODE": "OPTIMIZE_SIZE"
    // "CACHE_MODE": "OPTIMIZE_SPEED"
  }
}
```